_studio/shared/src/libmfx_allocator_vaapi.cpp

```diff
@@ -158,43 +232,7 @@ static mfxStatus ReallocImpl(mfxDefaultAllocatorVAAPI::mfxWideHWFrameAllocator*
 
     *vaapi_mid->m_surface = 0;
 
-    attrib[attrCnt].type            = VASurfaceAttribPixelFormat;
-    attrib[attrCnt].flags           = VA_SURFACE_ATTRIB_SETTABLE;
-    attrib[attrCnt].value.type      = VAGenericValueTypeInteger;
-    attrib[attrCnt++].value.value.i = va_fourcc;
-
-    format               = va_fourcc;
-
-    switch (fourcc)
-    {
-        case MFX_FOURCC_VP8_NV12:
-            // special configuration for NV12 surf allocation for VP8 hybrid encoder is required
-            attrib[attrCnt].type            = (VASurfaceAttribType)VASurfaceAttribUsageHint;
-            attrib[attrCnt].flags           = VA_SURFACE_ATTRIB_SETTABLE;
-            attrib[attrCnt].value.type      = VAGenericValueTypeInteger;
-            attrib[attrCnt++].value.value.i = VA_SURFACE_ATTRIB_USAGE_HINT_ENCODER;
-            break;
-        case MFX_FOURCC_VP8_MBDATA:
-            // special configuration for MB data surf allocation for VP8 hybrid encoder is required
-            attrib[0].value.value.i = VA_FOURCC_P208;
-            format                  = VA_FOURCC_P208;
-            break;
-        case VA_FOURCC_NV12:
-            format = VA_RT_FORMAT_YUV420;
-            break;
-        case VA_FOURCC_UYVY:
-        case VA_FOURCC_YUY2:
-            format = VA_RT_FORMAT_YUV422;
-            break;
-        case MFX_FOURCC_A2RGB10:
-            format = VA_RT_FORMAT_RGB32_10BPP;
-            break;
-        case MFX_FOURCC_RGBP:
-            format = VA_RT_FORMAT_RGBP;
-            break;
-        default:
-            break;
-    }
+    attrCnt = FillSurfaceAttrs(&attrib[0], &format,  fourcc, va_fourcc, 0);
```

Comment:
U've destroyed P8 buffer (`vaDestroyBuffer`) while reallocating, u're not creating new one though.

Reply:
In case of P8 `surfaces[0]` is destroyed as buffer (`vaDestroyBuffer`) and then it is also destroyed as surface (`vaDestroySurfaces`). It's logic error, isn't it?